src/Microsoft.AspNet.SignalR.Redis/RedisMessageBus.cs

```diff
@@ -27,6 +27,7 @@ public class RedisMessageBus : ScaleoutMessageBus
         private string _connectionString;
         private int _state;
         private readonly object _callbackLock = new object();
+        private readonly SemaphoreSlim _redisConnectionEventLock = new SemaphoreSlim(1, 1);
```

Comment:
I didn't feel it really makes a difference given that the other event handler always finishes very quickly and the collision is very unlikely.